File: stack/validParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for ch in s:
            if ch == "(" or ch == "[" or ch =="{":
                stack.append(ch)
            elif not len(stack):
                return False
            elif (ch == ")" and stack.pop() != "(") or (ch == "]" and stack.pop() != "[") or (ch == "}" and stack.pop() != "{"):
                return False
        return False if len(stack) else True

File: stack/validParentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        std::stack<char> stack;
        std::unordered_map<char, char> closeToOpen = {
            {')', '('},
            {'}', '{'},
            {']', '['}
        };
        for (char c : s){
            if(closeToOpen.count(c) == 0)
                stack.push(c);     
            else if (stack.empty() || stack.top() != closeToOpen[c])
                return false;
            else
                stack.pop();
        }
        return stack.empty();
    }
};
